tests/arch/arm/arm_thread_swap/src/arm_thread_arch.c

```diff
@@ -68,25 +68,31 @@ static void load_callee_saved_regs(const _callee_saved_t *regs)
 	__asm__ volatile (
 		"mov r1, r7;\n\t"
 		"mov r0, %0;\n\t"
+		"add r0, #16;\n\t"
 		"ldmia r0!, {r4-r7};\n\t"
+		"mov r8, r4;\n\t"
+		"mov r9, r5;\n\t"
+		"mov r10, r6;\n\t"
+		"mov r11, r7;\n\t"
+		"sub r0, #32;\n\t"
 		"ldmia r0!, {r4-r7};\n\t"
 		"mov r8, r4;\n\t"
 		"mov r9, r5;\n\t"
 		"mov r10, r6;\n\t"
 		"mov r11, r7;\n\t"
 		"mov r7, r1;\n\t"
```

Comment:
The second `ldmia` is loading `r4-r7` (i.e. `v1-v4`), so why are moving `r4-r7` to `r8-r11` here?

Reply:
The first one retrieves the upper 4 values into r4-7 (input[4]), then pushes them into r8-11, since `ldmia` can only operate on r0-7 on the M0, etc. The second one moves the pointer back to input[0], then places those four words into r4-7, where they belong.

Reply:
Ugg, sorry. I missed your point here. You're right, those lines needs to be removed.